api/envoy/extensions/router/cluster_specifiers/matcher/v3/matcher.proto

```diff
@@ -0,0 +1,25 @@
+syntax = "proto3";
+
+package envoy.extensions.router.cluster_specifiers.matcher.v3;
+
+import "xds/type/matcher/v3/matcher.proto";
+
+import "udpa/annotations/status.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.router.cluster_specifiers.matcher.v3";
+option java_outer_classname = "MatcherProto";
+option java_multiple_files = true;
+option go_package = "github.com/envoyproxy/go-control-plane/envoy/extensions/router/cluster_specifiers/matcher/v3;matcherv3";
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: Matcher Based Cluster Specifier]
+// [#extension: envoy.router.cluster_specifier_plugin.matcher]
```

Comment:
I assume this extension is meant to go into the `ClusterSpecifierPlugin` extension point in the route config?

https://github.com/envoyproxy/envoy/blob/0f5071fe474b8b4c0f2921edb1c79176f0370f39/api/envoy/config/route/v3/route_components.proto#L504

It looks like there's no extension category set up for this for documentation purposes.  It might be a good idea to add that, so that the docs can show which extensions work in this extension point.

Reply:
SGTM.